### ISSUE TITLE
Pokemon animations is not loading

### ISSUE THREAD

**Issue by SuadSeen** [Romulion/LMD_Animation_Loader#6]
I am trying to import a pokemon animation but is not loaded. But when i import a trainer Animation it’s working.

**Comment by Romulion** [comment]
Is this model from pokemon masters? (didnt see it on mega)
And what animation you trying to export?

**Comment by SuadSeen** [comment]
Yes, this model is from Pokemon masters ex. You can found pokemon models and animation of Pokemon Masters Ex from here https://mega.nz/folder/fOYCUDgb#PYMT02T-o2SC6328uivPcw. 
The trainer model animations are importing with your awesome plugin @Romulion, but the pokemon animations aren't working

**Comment by Romulion** [comment]
Animation has different rotaion mode - 2 vec3 arrays,
probably euler angle, but still cant get it look good.
Traslation part works well but has extra array with unknown purpose.
Althought it has UV animation and GRP (dont know what is this but it mostly packet whith zeros).

I`ll put it for later, and continue when something came to mind

**Comment by jeffw773** [comment]
> Animation has different rotaion mode - 2 vec3 arrays, probably euler angle, but still cant get it look good. Traslation part works well but has extra array with unknown purpose. Althought it has UV animation and GRP (dont know what is this but it mostly packet whith zeros).
> 
> I`ll put it for later, and continue when something come to mind

Any news on this at all? I am working on a full fledged Blender pack of all characters and pokemon setup with textures and animations to be released to the community and it would be nice to have animations for the pokemon working :)

So far I am 40% done with the pack for the characters, but haven't done the pokemon yet

**Comment by jeffw773** [comment]
> Animation has different rotaion mode - 2 vec3 arrays, probably euler angle, but still cant get it look good. Traslation part works well but has extra array with unknown purpose. Althought it has UV animation and GRP (dont know what is this but it mostly packet whith zeros).
> 
> I`ll put it for later, and continue when something come to mind

![image](https://user-images.githubusercontent.com/49732785/193184237-5ec6e473-e14b-4ccb-8af8-588b0ffe67ba.png)

I found this math which will likely help. I believe these pokemon animations use data from the Sun/Moon/ORAS games so this is the code that tools for converting those game's data to DAE format for animations so this might work!

Link to the tool and the page mentioned with the math in it: https://github.com/gdkchan/SPICA/blob/master/SPICA/Math3D/Vector.cs

**Comment by jeffw773** [comment]
![Math for rotations](https://user-images.githubusercontent.com/49732785/193184559-8e8d93a0-1454-40f7-9233-e56254e1b754.PNG)

**Comment by jeffw773** [comment]
![image](https://user-images.githubusercontent.com/49732785/193364761-9b9b411b-79e0-4e5f-897e-9fbf1cd954ea.png)

This has been my current marking up of voltorbs animation file since it is so small and simple.

**Comment by jeffw773** [comment]
![image](https://user-images.githubusercontent.com/49732785/193364801-d699f34f-e7e0-42ff-928f-7e27bd90d14d.png)

This is what is figured out so far

**Comment by jeffw773** [comment]
![ezgif com-gif-maker (83)](https://user-images.githubusercontent.com/49732785/194387210-ee73904e-2db2-4d87-b891-730034a96357.gif)
So far there is a working Pokemon Animation script created by user SleepyZay as of recently that works very well! Though it is in 3DS Max only right now

**Comment by jeffw773** [comment]
[SleepyZay_pokemon_masters_animation_2.zip](https://github.com/Romulion/LMD_Animation_Loader/files/9727438/SleepyZay_pokemon_masters_animation_2.zip)
This is what SleepyZay has figured out so far! Guy is wicked smart and deserves all credit for this file I attached. I am sure some of the offsets and data being read can be converted to Blender scripts somehow

**Comment by Romulion** [comment]
Thanx, it will greatly help, ill fix script tomorow

**Comment by Romulion** [comment]
Pokemon animation is ready (translation, rotation and scale).

**Comment by jeffw773** [comment]
Awesome! I will test it out!

**Comment by jeffw773** [comment]
> Pokemon animation is ready (translation, rotation and scale).

![image](https://user-images.githubusercontent.com/49732785/194601544-e08fcc73-5a49-4d69-994f-3a7aa11d60bc.png)
I am getting the animations coming in upside down haha

**Comment by jeffw773** [comment]
Nevermind! I reimported a fresh model and it works great!
